src/components/reviews/Reviews.jsx

```diff
@@ -110,37 +115,74 @@ class Reviews extends React.Component {
         reviewsList.push(review);
       }
     }
-    const reviewCount = reviewsList.length;
-    const moreToLoad = newReviews.length === this.state.reviewsPerLoad;
+    const reviewsLoaded = reviewsList.length;
+    const moreToLoad = !(newReviews.length < this.state.reviewsPerLoad);
     const filter = this.state.filter;
     const filteredReviewsList = [];
     for (let review of reviewsList) {
       if (filter(review)) {
         filteredReviewsList.push(review);
       }
     }
+    filteredReviewsList.sort(this.getSortFunction());
     this.setState(
-      { reviewsList, filteredReviewsList, reviewCount, moreToLoad, isLoading: false },
+      { reviewsList, filteredReviewsList, reviewsLoaded, moreToLoad, isLoading: false },
       callback
     );
   }
 
+  getSortFunction() {
+    const { sortOrder } = this.state;
+    if (sortOrder === 'newest') {
+      return (a, b) => {
+        if (new Date(a.date) < new Date(b.date)) {
+          return 1;
+        } else {
+          return -1;
+        }
+      };
+    } else if (sortOrder === 'relevant') { // not sure how to sort for relevancy?
+      return (a, b) => {};
+    } else if (sortOrder === 'helpful') {
+      return (a, b) => {
+        if (new Date(a.helpfulness) < new Date(b.helpfulness)) {
```

Comment:
I've never seen date used this way before, is "helpfulness" formatted in some way for you that makes it valid for Date() and I'm just missing it?